eve-tools/bpftrace-compiler/edgeview.go

```diff
@@ -0,0 +1,255 @@
+// Copyright (c) 2024 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io/fs"
+	"log"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+	"unicode"
+
+	"github.com/creack/pty"
+)
+
+type edgeviewRun struct {
+	cmd    *exec.Cmd
+	ctx    context.Context
+	cancel func()
+	path   string
+	port   uint16
+	tty    *os.File
+}
+
+func pathToScript(path string) string {
+	scripts, err := filepath.Glob(path)
+	if err != nil {
+		log.Fatalf("could not glob '%s': %v", path, err)
+	}
+
+	newestScript := ""
+	newestTime := time.Unix(0, 0)
+	for _, script := range scripts {
+		st, err := os.Stat(script)
+		if err != nil {
+			continue
+		}
+
+		if st.ModTime().After(newestTime) {
+			newestTime = st.ModTime()
+			newestScript = script
+		}
+	}
+
+	return newestScript
+}
+
+func (er edgeviewRun) enablePprof() error {
+	edgeviewParam := "pprof/on"
+	matchOutput := "=== System: <pprof/on> ==="
+
+	_, err := er.runEdgeview(edgeviewParam, matchOutput)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (er edgeviewRun) disablePprof() error {
+	edgeviewParam := "pprof/off"
+	matchOutput := "=== System: <pprof/off> ==="
+
+	_, err := er.runEdgeview(edgeviewParam, matchOutput)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (er *edgeviewRun) stopHTTPDebugPort() {
+	_, err := er.tty.Write([]byte{'\x03'})
+	if err != nil {
+		_, ok := err.(*fs.PathError)
+		if !ok {
+			log.Printf("could not send ctrl+c: %v", err)
+		}
+	}
+	if er.cmd != nil {
+		_ = er.cmd.Process.Signal(os.Interrupt)
+	}
+
+	if er.cancel != nil {
+		er.cancel()
+		er.ctx = nil
+		er.port = 0
+	}
+	if er.cmd != nil {
+		err = er.cmd.Wait()
+		if err != nil {
+			exitErr, ok := err.(*exec.ExitError)
+			if ok {
+				status := exitErr.Sys().(syscall.WaitStatus)
+				if !status.Signaled() {
+					log.Printf("expected that the process exited because of signal, but %v", err)
+				}
+			} else if !strings.Contains(err.Error(), "Wait was already called") {
+				log.Printf("waiting failed: %v/%T", err, err)
+			}
+		}
+	}
+	err = er.tty.Close()
+	if err != nil {
+		_, ok := err.(*fs.PathError)
+		if !ok {
+			log.Printf("closing tty failed: %v", err)
+		}
+	}
+
+	if er.cmd != nil {
+		_ = er.cmd.Process.Kill()
+		er.cmd = nil
+	}
+}
+
+func (er *edgeviewRun) forwardHTTPDebugPort() error {
+	var err error
+
+	er.ctx, er.cancel = context.WithCancel(context.Background())
+	er.cmd = exec.CommandContext(er.ctx, "bash", er.path, "tcp/127.0.0.1:6543")
+
+	er.tty, err = pty.Start(er.cmd)
+	if err != nil {
+		log.Fatalf("starting pty for cmd (%v) failed: %v", er.cmd, err)
+	}
+
+	matchedChan := make(chan struct{})
+
+	var forwardLine string
+	go func() {
+
+		var output string
```

Comment:
Do we need output here? I don't see we use it later.

Reply:
Thanks.
I will add it to the returned `err`.